### PR TITLE
Windows: Enable issue 70093 link tests

### PR DESCRIPTION
**Pull request by ChrisDenton**
Tracking issue for `-Z link-native-libraries`: #134948
Tracking issue for `-Z link-directives`: #134947

`-Zlink-native-libraries=no` and `-Zlink-directives=no` *should* work on Windows, at least for msvc. The fly in ointment is that `default-linker-libraries` doesn't. On unixy platforms rustc calls another compiler which in turn calls the linker along with the default libraries. On MSVC rustc calls the linker directly therefore it would need to be the one to implement `default-linker-libraries`. Except it doesn't so we workaround that in the test by using `-C link-arg` to talk to the linker.

### CODE REVIEW

**Comment by ChrisDenton** [comment]
All tests pass on windows-msvc and windows-gnu hosts, marking as ready for review.

**Comment by saethlin** [comment]
Can you do this with revisions like in https://github.com/rust-lang/rust/blob/2061630860906083681a32ed748e53bee4a03fb6/tests/ui/repr/repr-c-dead-variants.rs ?

**Comment by ChrisDenton** [comment]
I can try but I'm uncertain if that will work due to cross-compiling.

**Comment by ChrisDenton** [comment]
That worked, thanks!

File: tests/ui/link-native-libs/issue-70093/link-native-libraries.rs
Comment (jieyouxu) [inline tests/ui/link-native-libs/issue-70093/link-native-libraries.rs:1]: Suggestion: could we slightly document what this test is actually checking?

I opened two tracking issues:

- https://github.com/rust-lang/rust/issues/134947
- https://github.com/rust-lang/rust/issues/134948

But the original issue https://github.com/rust-lang/rust/issues/70093 is not very obvious in what this test is actually trying to check for.

**Review by jieyouxu — COMMENTED** [review]
Thanks, one request for a test doc comment, but otherwise LGTM.

**Review by jieyouxu — APPROVED** [review]

File: tests/ui/link-native-libs/issue-70093/link-native-libraries.rs
Comment (ChrisDenton) [reply tests/ui/link-native-libs/issue-70093/link-native-libraries.rs:1]: These tests were the first time I'd heard of `link-directives` and `link-native-libraries` but I think I get the intent. Let me see what I can write.

Comment (jieyouxu) [reply tests/ui/link-native-libs/issue-70093/link-native-libraries.rs:1]: Same for me, which is how I found out they had no MCPs nor tracking issues, and `-Zlink-directives` I believe has zero documentation either.

Comment (ChrisDenton) [reply tests/ui/link-native-libs/issue-70093/link-native-libraries.rs:1]: Tbh, I think they have a lot of overlap with each other. And also with using the `-l` compiler flag when used to override native libs (but that currently has a much more limited effect).

**Review by jieyouxu — APPROVED** [review]
Thanks, you can r=me after PR CI is green.

**Pull request merged**